modules/videoio/cmake/detect_ffmpeg.cmake

```diff
@@ -99,6 +99,38 @@ if(HAVE_FFMPEG_WRAPPER)
   ocv_add_external_target(ffmpeg "" "" "HAVE_FFMPEG_WRAPPER")
 elseif(HAVE_FFMPEG)
   ocv_add_external_target(ffmpeg "${FFMPEG_INCLUDE_DIRS}" "${FFMPEG_LIBRARIES}" "HAVE_FFMPEG")
+  set(__builtin_defines "")
+  set(__builtin_include_dirs "")
+  set(__builtin_libs "")
+  set(__plugin_defines "")
+  set(__plugin_include_dirs "")
+  set(__plugin_libs "")
+  if(HAVE_OPENCL)
+    set(__opencl_dirs "")
+    if(OPENCL_INCLUDE_DIRS)
+      set(__opencl_dirs "${OPENCL_INCLUDE_DIRS}")
+    elseif(OPENCL_INCLUDE_DIR)
+      set(__opencl_dirs "${OPENCL_INCLUDE_DIR}")
+    else()
+      set(__opencl_dirs "${OpenCV_SOURCE_DIR}/3rdparty/include/opencl/1.2")
```

Comment:
I observe cmake error in some configurations:
```
CMake Error in modules/CMakeLists.txt:
  Target "ocv.3rdparty.ffmpeg.builtin_deps" INTERFACE_INCLUDE_DIRECTORIES
  property contains path:

    "/opencv/3rdparty/include/opencl/1.2"

  which is prefixed in the source directory.
```
It seem to be related to installation process: https://stackoverflow.com/questions/25676277/cmake-target-include-directories-prints-an-error-when-i-try-to-add-the-source

Reply:
Thanks, this problem is caused by `-DBUILD_SHARED_LIBS=OFF` option.